### PR TITLE
add kubeconfig flag; fix for issue#2378

### PR DESCRIPTION
**Pull request by rajatchopra**
fix for issue #2378

Note that the kubeconfig will not be overridden if the config is borrowed from kubectl env (though that code is disabled right now).

Signed-off-by: Rajat Chopra <rchopra@redhat.com>

### CODE REVIEW

File: cmd/kops/root.go [hunk 1]
```diff
@@ -131,6 +135,9 @@ func NewCmdRoot(f *util.Factory, out io.Writer) *cobra.Command {
 	defaultClusterName := os.Getenv("KOPS_CLUSTER_NAME")
 	cmd.PersistentFlags().StringVarP(&rootCommand.clusterName, "name", "", defaultClusterName, "Name of cluster. Overrides KOPS_CLUSTER_NAME environment variable")
 
+	// file paths are common to all sub commands
+	cmd.PersistentFlags().StringVar(&rootCommand.pathOptions.LoadingRules.ExplicitPath, rootCommand.pathOptions.ExplicitFileFlag, rootCommand.pathOptions.LoadingRules.ExplicitPath, "use a particular kubeconfig file")
```
Comment (andrewsykim) [inline cmd/kops/root.go]: Can we document the default behavior here?

**Comment by justinsb** [comment]
So `--kubeconfig` is now on every command, because you added the flag to the root.  I'm not sure whether that's a good thing or not.

One option would be to add it just to `kops export kubecfg` for now, and then if we decide it should be on other commands, we can add it to those or promote it to the root.

**Comment by justinsb** [comment]
Thank you for the PR @rajatchopra, but I'm going to move this to the next milestone; I think we should try the lower impact option first

**Comment by justinsb** [comment]
We discussed in (office-hours today)[https://www.youtube.com/watch?v=SdW4Y7FgZfI&index=2&list=PL69nYSiGNLP2NcuGKoAzme6ZtZmrJlsaR&t=930s] the need to prioritize bug fixes until we can get 1.12 out (which will hopefully mean we get to everything faster overall), and I believe this is a feature, so I'm going to move this to the next release.

**Pull request closed without merging**